crates/router/src/core/user.rs

```diff
@@ -1093,13 +1097,24 @@ pub async fn switch_merchant_id(
             .organization_id;
 
         let token = utils::user::generate_jwt_auth_token_with_custom_role_attributes(
-            state,
+            &state,
             &user,
             request.merchant_id.clone(),
-            org_id,
+            org_id.clone(),
             user_from_token.role_id.clone(),
         )
         .await?;
+
+        utils::user_role::set_role_permissions_in_cache_if_required(
+            &state,
+            user_from_token.role_id.as_str(),
+            request.merchant_id.as_str(),
+            org_id.as_str(),
+        )
+        .await
+        .map_err(|e| logger::error!("Error setting permissions in cache {:?}", e))
+        .ok();
```

Comment:
not required since internal user's permissions will always be in predefined permissions.